libraries/rundeck_api_client.rb

```diff
@@ -3,10 +3,14 @@
 require 'openssl'
 
 class RundeckApiClient
-  def initialize(rundeck_server_url, user, opts={})
+  def initialize(rundeck_server_url, user, pass, opts={})
     @rundeck_server_url, @user, = rundeck_server_url, user
+
     # convert all opts keys to symbols
     @opts = opts.each_with_object({}) { |(k,v), h| h[k.to_sym] = v }
+
+    # the api only responds to authenticated clients, so call it from the constructor
```

Comment:
normally id be against adding logic / functionality like this within a constructor, but i think it makes sense here because without it everytime you'd have to write

```ruby
client = RundeckApiClient.new 'https://rundeck.url', 'username'
client.authenticate 'password'
client.get '/path'
```

but with this you just write
```ruby
client = RundeckApiClient.new 'https://rundeck.url', 'username', 'password'
client.get '/path'
```